src/SignalR/common/Shared/SystemTextJsonExtensions.cs

```diff
@@ -80,7 +78,7 @@ public static bool ReadAsBoolean(this ref Utf8JsonReader reader, string property
                 throw new InvalidDataException($"Expected '{propertyName}' to be of type {JsonTokenType.String}.");
             }
 
-            return reader.GetString();
+            return reader.GetString()!;
```

Comment:
Now that's just a lie

Reply:
No it isn't, we check above that the token type is a string. If the string is null it would have token type null.